### ISSUE TITLE
Average quality of reads

### ISSUE THREAD

**Issue by dariober** [wdecoster/nanomath#14]
Hi- based on the reasoning that base quality cannot be directly averaged because they are in log space ([here](https://gigabaseorgigabyte.wordpress.com/2017/06/26/averaging-basecall-quality-scores-the-right-way/)), shouldn't the overall average read quality be converted to probabilities, averaged and converted back to phred score? 

I mean, in [nanomath.py#L43](https://github.com/wdecoster/nanomath/blob/master/nanomath/nanomath.py#L43) there is

```
self.mean_qual = np.mean(df["quals"])
```

shouldn't it be:

```
self.mean_qual = -10 * log(sum(10**(q/-10) for q in df["quals"]]) / len(df["quals"]), 10)
```

Thanks - Dario

**Comment by wdecoster** [comment]
Hi Dario,

TLDR: I don't think so, but I'm not too sure about it, and may have to think more about it.

The way I look at this is:
- if I want to know the average quality of a read then I need to take into account that I need to convert to probabilities and back again, because just adding and dividing does the wrong thing when aggregating. This is essentially summarizing the qualities to get one quality for the read - to get an idea of how the entire read behaved. It tells you little about individual bases within the read, I guess.
- however, if I want to know how the average read performed for any metric, I think a normal average suffices. I want to know how well the average molecule in my mix is doing.

But I must admit I haven't thought all that long about this, and I'd definitely be interested in hearing more opinions.

Medians are probably better than averages anyway :)

Cheers,
Wouter
